### PR TITLE
Proposed changes to API

### PR DESCRIPTION
**Pull request by RichRick1**
1. made load_data, saveoutput and _normalize_desc method the concrete ones, because they shouldn't depend on the cluster or diversity method of selection
2. add __init__ method to a SelectionBase abstract class, because there are properties that can be the same for both Cluster and Dissimilarity selectors
3. rewrite DessimilaritySelector class based on the changes of the parent class (even thought it's an abstract class)

@fwmeng88 any suggestions on how to make it better?

### CODE REVIEW

**Review by fwmeng88 — CHANGES_REQUESTED** [review]
Thanks for the contribution! There are just some minor changes required or something that I am not very clear of.

File: DiverseSelector/base.py [hunk 1]
```diff
@@ -23,32 +23,52 @@
...
+    @abstractmethod  # abstract method, because we want in to be in both child classes
     def select(self):
         """Select the subset molecules with optimal diversity."""
         pass
 
     @property
-    @abstractmethod
-    def subset_diversity(self):
+    def subset_diversity(self):  # concrete method, because we want in to be in both child classes, and it should act
+        # in the same way
```
[the remark this answers is not in the document]
Comment (RichRick1) [reply DiverseSelector/base.py:63]: You are right. Idea here is if we gonna use the same structure of the data and the same method for calculating diversity of the subset, we can use one method for calculating the subset diversity and code it here _explicitly_

File: DiverseSelector/base.py [hunk 2]
```diff
@@ -23,32 +23,52 @@
...
         """
         Calculate diversity of the subset."""
         # todo: need to implement diversity measurement here
         pass
 
     @property
-    @abstractmethod
-    def all_diversity(self):
+    def all_diversity(self):  # concrete method, because we want in to be in both child classes, and it should act
+        # in the same way
```
[the remark this answers is not in the document]
Comment (RichRick1) [reply DiverseSelector/base.py:71]: The same idea. As long as we are having all data in the same format and all methods for calculating diversity are the same at both DiverseSelector and ClusterSelector, we can make this method _concrete_ and implement right in here

File: DiverseSelector/dissimilarity_based.py [hunk 3]
```diff
@@ -103,45 +90,3 @@ def select(self):
         # todo: mimmax algorithm implementation
         # for idx_counter in np.arange(1, self.num_selected):
         pass
```
[the remark this answers is not in the document]
Comment (RichRick1) [reply DiverseSelector/dissimilarity_based.py:92]: Sure! Will do

**Comment by RichRick1** [comment]
Implement load_data as moving get_features

**Comment by RichRick1** [comment]
@fwmeng88 add implementation of MinMax algorithm and fixed `load_data` function. 
Also, add visualization for MinMax algorithm, however faced several issues with relative imports, so in visualization only code with the algorithm
Please, let me know what do you think about it

**Review by fwmeng88 — CHANGES_REQUESTED** [review]
Thanks for the nice improvement! Only one minor change for the imports. @RichRick1

File: DiverseSelector/base.py [hunk 4]
```diff
@@ -23,32 +23,44 @@
...
 """Base class for diversity based subset selection."""
 
-from abc import ABC, abstractmethod, ABCMeta
-
-__all__ = [
-    "DissimilaritySelectionBase",
-    "ClusteringSelectionBase",
-]
+from abc import ABC, abstractmethod
+from utils import *
```
[the remark this answers is not in the document]
Comment (RichRick1) [reply DiverseSelector/base.py]: Thanks! It's been fixed

**Review by fwmeng88 — APPROVED** [review]
Thanks for the nice contribution!!

**Pull request merged**